modules/ocf/manifests/auth.pp

```diff
@@ -132,11 +132,23 @@
     require => Augeas['sshd_config'],
   }
 
+  # Get all DNS names, FQDNs, and IPs for a host to include in SSH keys
+  $ssh_aliases = delete(concat(
+    suffix(delete(any2array($::dnsA), ''), ".${::domain}"),
+    $::dnsA,
+    suffix(delete(any2array($::dnsCname), ''), ".${::domain}"),
+    $::dnsCname,
+    $::fqdn,
+    $::ipHostNumber,
+    $::ipaddress6,
+  ), '')
```

Comment:
I think there should be a nicer way to work with the puppet stdlib package to do this, but I couldn't find a way since `$::dnsA` and `$::dnsCname` can be empty, have a single element, or be an array, so there doesn't seem like an easy way to work with them except converting them both to arrays, and then removing any empty strings that still exist.

Reply:
Maybe the ldap-enc script could be modified to handle this better, so that we tell it which values are array-type and which are single element only?